lib/compiler/src/engine/artifact.rs

```diff
@@ -931,7 +1159,7 @@ impl Artifact {
 
         Ok(Self {
             id: Default::default(),
-            artifact,
+            artifact: ArtifactBuildVariant::Plain(artifact),
             allocated: Some(AllocatedArtifact {
                 frame_info_registered: false,
                 frame_info_registration: None,
```

Comment:
Let's take this opportunity to add some sorely needed focued Artifact tests.

I don't know if the all the related types implement (Partial)Eq, but we could have a test that round-trips a module ,and then compares the two variants for equality.

We should also have some fixtures of serialized modules in files, and test that they can still be loaded so we don't accidentally break deserialization.

Note that there is an artifact version number that we currently manually increment if the layout changes, which is also used for constructing the artifact cache filenames.

https://github.com/wasmerio/wasmer/blob/7602de387cd6a4cc940098c170320f9f5229a847/lib/types/src/serialize.rs#L214

Reply:
Tests should use some reasonably complex modules, like Python/PHP. 

I think there are already `.wasm` files in the repo, or if not, they can be extracted from the webcs.

Reply:
I added two new tests, one for serializing freshly-compiled modules and one for deserializing existing compiled modules.

Reply:
Awesome, thanks.